tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -294,27 +295,52 @@ public void Authentication_succeeds_when_user_has_multiple_credentials_and_mecha
             var settings = client.Settings.Clone();
             settings.Credential = MongoCredential.FromComponents(
                 mechanism: "SCRAM-SHA-256", source: null, username: uniqueUnicodeUserName, password: unicodePassword);
-            
+
             AssertAuthenticationSucceeds(settings);
         }
 #endif
-        
+
+        [SkippableFact]
+        public void Authentication_succeeds_with_MONGODB_X509_mechanism()
+        {
+            RequireServer.Check().Tls(required: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH", isDefined: true);
+            RequireEnvironment.Check().EnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD", isDefined: true);
+            var pathToClientCertificate = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PATH");
+            var password = Environment.GetEnvironmentVariable("MONGO_X509_CLIENT_CERTIFICATE_PASSWORD");
+            var clientCertificate = new X509Certificate2(pathToClientCertificate, password);
+            var userName = GetRfc2253FormattedUsernameFromX509ClientCertificate(clientCertificate);
+            DropDatabaseUser(DriverTestConfiguration.Client, userName, databases: new[] { "$external" });
+            CreateX509DatabaseUser(DriverTestConfiguration.Client, userName);
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            var serverVersion = CoreTestConfiguration.ServerVersion;
+
+            var serverExtractsUsernameFromCertificate =
+                serverVersion >= Feature.ServerExtractsUsernameFromX509Certificate.FirstSupportedVersion;
+            settings.Credential = MongoCredential.CreateMongoX509Credential(
+                username: serverExtractsUsernameFromCertificate ? null : userName);
```

Comment:
We need this because we test against some pretty ancient server versions in EG :)